src/module_loader.rs

```diff
@@ -14,19 +14,51 @@ use std::{
     rc::Rc,
 };
 
+#[allow(unused_variables)]
+#[cfg(feature = "custom_import")]
+pub trait ImportProvider {
+    fn resolve(
+        &mut self,
+        specifier: &str,
+        referrer: &str,
+        kind: deno_core::ResolutionKind,
+    ) -> Result<ModuleSpecifier, anyhow::Error> {
+        Ok(deno_core::resolve_import(specifier, referrer)?)
+    }
+    fn import(
+        &mut self,
+        specifier: ModuleSpecifier,
+        referrer: Option<ModuleSpecifier>,
+        is_dyn_import: bool,
+        requested_module_type: deno_core::RequestedModuleType,
+    ) -> Result<String, anyhow::Error> {
+        return Err(anyhow!(
+            "unrecognized schema for module import: {specifier}"
+        ));
+    }
+}
+
 type SourceMapCache = HashMap<String, (String, Vec<u8>)>;
 
 #[derive(Clone)]
 struct InnerRustyLoader {
     cache_provider: Rc<Option<Box<dyn ModuleCacheProvider>>>,
+    import_provider: Rc<Option<RefCell<Box<dyn ImportProvider>>>>,
     fs_whlist: Rc<RefCell<HashSet<String>>>,
     source_map_cache: Rc<RefCell<SourceMapCache>>,
 }
 
 impl InnerRustyLoader {
-    fn new(cache_provider: Option<Box<dyn ModuleCacheProvider>>) -> Self {
+    fn new(
+        cache_provider: Option<Box<dyn ModuleCacheProvider>>,
+        import_provider: Option<Box<dyn ImportProvider>>,
+    ) -> Self {
         Self {
             cache_provider: Rc::new(cache_provider),
+            import_provider: Rc::new(match import_provider {

```

Comment:
Would this be cleaner with import_provider.map?

Reply:
Could be. Will try it out.